code/modules/mob/living/carbon/carbon_update_icons.dm

```diff
@@ -79,14 +79,24 @@
 	apply_overlay(HANDS_LAYER)
 
 
-/mob/living/carbon/update_fire(fire_icon = "Generic_mob_burning")
+/mob/living/carbon/update_fire()
 	remove_overlay(FIRE_LAYER)
+	remove_overlay(SECONDARY_FIRE_LAYER)
+
 	if(on_fire || HAS_TRAIT(src, TRAIT_PERMANENTLY_ONFIRE))
-		var/mutable_appearance/new_fire_overlay = mutable_appearance('icons/mob/OnFire.dmi', fire_icon, -FIRE_LAYER)
-		new_fire_overlay.appearance_flags = RESET_COLOR
-		overlays_standing[FIRE_LAYER] = new_fire_overlay
+		var/fire_override = dna?.species?.fire_overlay
```

Comment:
```suggestion
		var/fire_override = dna?.species.fire_overlay
```
DNA should never be null here but sanity is fine here. Species will NEVER be null if dna is present I don't believe.